sched/sched/sched_critmonitor.c

```diff
@@ -280,7 +280,6 @@ void nxsched_resume_critmon(FAR struct tcb_s *tcb)
       /* Yes.. Save the start time */
 
       tcb->premp_start = current;
-      DEBUGASSERT(tcb->premp_start != 0);
```

Comment:
Do we know why assertion is here and why it is removed?
The PR description does not reflect this info

Reply:
@pkarashchenko 

Perf timer interface generally uses the hardware cycle counter
provided by the arch chip directly(such as DWT_CYCCNT(cortex-m)),
CYCCNT is a free running counter and counting upwards.
It wraps around to 0 on overflow.

I have update the description on commit message, please review again